oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/IndexUpdate.java

```diff
@@ -276,9 +276,19 @@ private void collectIndexEditors(NodeBuilder definitions,
                     rootState.corruptIndexHandler.skippingCorruptIndex(rootState.async, indexPath, ISO8601.parse(corruptSince));
                     continue;
                 }
-
-                Editor editor = rootState.provider.getIndexEditor(type, definition, rootState.root,
-                        rootState.newCallback(indexPath, shouldReindex, getEstimatedCount(definition)));
+                Editor editor = null;
+                try {
+                    editor = rootState.provider.getIndexEditor(type, definition, rootState.root,
+                            rootState.newCallback(indexPath, shouldReindex, getEstimatedCount(definition)));
+                } catch (IllegalStateException e) {
```

Comment:
Would it be possible to log the problem where the IllegalStateException is thrown, that is in the IndexDefinition.java file? Maybe in the builder... I just think that the IndexUpdate class is more far away than necessary from there the exception is thrown. I think it is possible to move it closer to index definition building. Still it might re-throw an exception. It's just that in the future, there might be other reasons why IllegalStateException that are not related to a wrong index definition itself (maybe for a remote index).

Reply:
@thomasmueller  - Sure - I will log the problem in IndexDefinition . I will rethrow the exception from there - and in the catch in the IndexUpdate  and just skip the iteration for this index (without logging the error)